internal/bminventory/inventory.go

```diff
@@ -5157,6 +5145,26 @@ func (b *bareMetalInventory) getBoundCluster(db *gorm.DB, infraEnv *common.Infra
 	return nil, nil
 }
 
+// Checks whether the specified obj can be updated by the user.
+// If not allowed, checks whether it can be read by the user.
+// This is required in order to return an appropriate error for
+// users who are only allowed to read.
+func (b *bareMetalInventory) checkAccessToObj(ctx context.Context, obj interface{}, objType string, objId *strfmt.UUID) error {
+	canWrite, err1 := b.authzHandler.HasAccessTo(ctx, obj, auth.UpdateAction)
+	if !canWrite {
+		canRead, err2 := b.authzHandler.HasAccessTo(ctx, obj, auth.ReadAction)
+		if !canRead {
+			msg := fmt.Sprintf("Failed to find %s with ID %s", objType, objId)
```

Comment:
We need to make sure to return the same exact message as with a "real" 404 error, so that users don't know if it's not there or if there are insufficient permissions

Reply:
Good point. Changed to "Object Not Found"